docs/blocks/index.md

```diff
@@ -1,51 +1,61 @@
 :::
 
-### Advanced Blocks
+### More functionality
 
-Of course, the above only allows for extremely basic blocks. If you want to add functionality, like player interaction, a custom class is required. However, the `Block` class has many methods and unfortunately not every single one can be documented here. See the rest of the pages in this section for things you can do with blocks.
+Of course, the above only allows for extremely basic blocks. If you want to add functionality, like player interaction, a custom class that extends `Block` is required. The `Block` class has many methods and unfortunately, not every single one can be documented here. Please refer to other pages in this section, or to the code of `Block`, for more information.
 
-Registering a Block
--------------------
+If you want to make a block that has different variants (think a slab that has a bottom, top, and double variant), you should use [blockstates]. And finally, if you want a block that stores additional data (think a chest that stores its inventory), a [block entity] should be used.
 
-Blocks must be [registered][registering] to function.
+### Resources
 
-:::caution
-A block in the level and a "block" in an inventory are very different things. A block in the level is represented by an `BlockState`, and its behavior defined by an instance of `Block`. Meanwhile, an item in an inventory is an `ItemStack`, controlled by an `Item`. As a bridge between the different worlds of `Block` and `Item`, there exists the class `BlockItem`. `BlockItem` is a subclass of `Item` that has a field `block` that holds a reference to the `Block` it represents. `BlockItem` defines some of the behavior of a "block" as an item, like how a right click places the block. It's possible to have a `Block` without an `BlockItem`. (E.g. `minecraft:water` exists a block, but not an item. It is therefore impossible to hold it in an inventory as one.)
+If you register your block and place it in the world, you will find it to be missing things like a texture. This is because textures, among others, are handled by Minecraft's resource system.
 
-When a block is registered, *only* a block is registered. The block does not automatically have an `BlockItem`. To create a basic `BlockItem` for a block, one should set the registry name of the `BlockItem` to that of its `Block`. Custom subclasses of `BlockItem` may be used as well. Once an `BlockItem` has been registered for a block, `Block#asItem` can be used to retrieve it. `Block#asItem` will return `Items#AIR` if there is no `BlockItem` for the `Block`, so if you are not certain that there is an `BlockItem` for the `Block` you are using, check for if `Block#asItem` returns `Items#AIR`.
-:::
+To apply a simple texture to a block, you must add a blockstate JSON, a model JSON, and a texture. See the section on [resources] for more information.
```

Comment:
Probably should link to the actual resource page or MC wiki links. This doesn't tell me much.

Reply:
Linking to the Minecraft wiki doesn't explain where your files go either, it just explains the general format. I put the link to the Resources section in here for now; I'm gonna try and rework Resources after this is done, I just didn't wanna do that in the same PR, as it'd get messy.

Reply:
True, but it's better than nothing. Splitting it into separate PRs is probably for the best. Most of the docs need to be rewritten anyway, so I'll take what you're willing to do. I'll cover the rest once I have my hands on my new laptop.